src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -180,11 +180,20 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
             }
 
             var constructorArguments = analyzedArguments.ConstructorArguments;
-            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.ToImmutableAndFree();
+
+            // If the arguments didn't have a natural type, they're not constants, and so would already have diagnostics from the rest of binding.
+            var ignoredDiagnostics = new DiagnosticBag();
+            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.SelectAsArray(
```

Comment:
>constructorArguments.Arguments [](start = 72, length = 30)

AlekseyTs said:

> Are we leaking this builder?

Reply:
Changing this back to resolved so you see my update here: `AnalyzedArguments.Arguments` are not pooled. There is no need to free that builder.

---
In reply to: [575600283](https://github.com/dotnet/roslyn/pull/51206#discussion_r575600283) [](ancestors = 575600283)